evm/src/test/java/org/hyperledger/besu/evm/operations/AuthOperationsTest.java

```diff
@@ -0,0 +1,110 @@
+import org.hyperledger.besu.evm.gascalculator.PragueGasCalculator;
+import org.hyperledger.besu.evm.operation.AuthCallOperation;
+import org.hyperledger.besu.evm.operation.AuthOperation;
+import org.hyperledger.besu.evm.worldstate.WorldUpdater;
+
+import java.util.Optional;
+
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.bytes.Bytes32;
+import org.apache.tuweni.units.bigints.UInt256;
+import org.junit.jupiter.api.Test;
+
+public class AuthOperationsTest {
+
+  @Test
+  public void testAuthOperation() {
+    SignatureAlgorithm algo = SignatureAlgorithmFactory.getInstance();
+    KeyPair keys = algo.generateKeyPair();
+    Address authingAddress = Address.extract(keys.getPublicKey());
+    EVM fakeEVM = mock(EVM.class);
+
+    Optional<Bytes> chainId = Optional.of(Bytes.of(1));
+    when(fakeEVM.getChainId()).thenReturn(chainId);
+    long senderNonce = 0;
+    Address invokerAddress = Address.fromHexString("0xdeadbeef");
+    Bytes32 invoker = Bytes32.leftPad(invokerAddress);
+    Bytes32 contractCommitment = Bytes32.leftPad(Bytes.fromHexString("0x1234"));
+    Bytes authPreImage =
+        Bytes.concatenate(
+            Bytes.ofUnsignedShort(AuthOperation.MAGIC),
+            chainId.get(),
+            Bytes32.leftPad(Bytes.ofUnsignedLong(senderNonce)),
+            invoker,
+            contractCommitment);
+    Bytes32 messageHash = Hash.keccak256(authPreImage);
+    SECPSignature signature = algo.sign(messageHash, keys);
+
+    MessageFrame frame = mock(MessageFrame.class);
+    when(frame.getContractAddress()).thenReturn(invokerAddress);
+    MutableAccount authingAccount = mock(MutableAccount.class);
+    when(authingAccount.getAddress()).thenReturn(authingAddress);
+    when(authingAccount.getNonce()).thenReturn(senderNonce);
+
+    WorldUpdater state = mock(WorldUpdater.class);
+
+    when(state.getAccount(authingAddress)).thenReturn(authingAccount);
+
+    when(frame.getWorldUpdater()).thenReturn(state);
+
+    when(frame.getSenderAddress()).thenReturn(authingAddress);
+    when(state.getSenderAccount(frame)).thenReturn(authingAccount);
+    when(frame.getStackItem(0)).thenReturn(authingAddress);
+    when(frame.getStackItem(1)).thenReturn(Bytes.of(0));
+    when(frame.getStackItem(2)).thenReturn(Bytes.of(97));
+    Bytes encodedSignature = signature.encodedBytes();
+    when(frame.readMemory(0, 1)).thenReturn(encodedSignature.slice(64, 1));
+    when(frame.readMemory(1, 32)).thenReturn(Bytes32.wrap(encodedSignature.slice(0, 32).toArray()));
+    when(frame.readMemory(33, 32))
+        .thenReturn(Bytes32.wrap(encodedSignature.slice(32, 32).toArray()));
+    when(frame.readMemory(65, 32)).thenReturn(contractCommitment);
+
+    AuthOperation authOperation = new AuthOperation(new PragueGasCalculator());
+    authOperation.execute(frame, fakeEVM);
+    verify(frame).setAuthorizedBy(authingAddress);
+    verify(frame).pushStackItem(UInt256.ONE);
+  }
+
+  @Test
+  public void authCallHappyPath() {
```

Comment:
* Needs failure case
* Both need to assert the top stack value is 1 (success) or 0 (failure)